ruby_event_store-rom/lib/ruby_event_store/rom.rb

```diff
@@ -10,10 +10,11 @@
 module RubyEventStore
   module ROM
     class Env
-      attr_accessor :container
+      attr_accessor :container, :lock
   
       def initialize(container)
         @container = container
+        @lock = Mutex.new
```

Comment:
http://railseventstore.org/docs/expected_version/ 

the purpos of `:any` expected version (which is default) is to NEVER be blocking for the client so that they can have maximal possible throughput, so that if you add RES to your project you don't experience locks or performance issues. This limits concurrency on threaded servers such as puma. It seems to be a workaround instead of addressing the actual issue imho.

Reply:
Ah, you're probably referring to how `Mutex` degrades `:any` use case.

Reply:
Indeed, but good point!